### ISSUE TITLE
Suggestion for smart update rework

### ISSUE THREAD

**Issue by Xori71** [mihonapp/mihon#144]
### Describe your suggested feature

Smart update would have an automatic setting (basically the one currently implemented) and a manual one.

![Screenshot_20240120-014829](https://github.com/mihonapp/mihon/assets/101265862/b677d547-62c8-4bff-953e-e3e3e1d1dd7a)

Users would be able to manually set the number of days until the next update. Then Mihon would set the difference of the update date and the last chapter’s upload date as the update schedule.

The benefits are more accurate update schedules, and easy mitigation of changes in manga schedules. Furthermore, multiple chapters with the same upload date (generally volume uploads) would no longer affect the schedule.

There is one problem with this: users could potentially exploit this function and make daily updates for manga a thing. This is why I also suggest that should the above be implemented, an internal app process should also be, which would work like this: It would keep count of how many updates yielded zero results for each manga in the library, and after a set number of zero-yield updates, the manga schedule would automatically revert to the automatic one calculated by Mihon.

### Other details

None.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an official extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/extensions/issues/new/choose).
- [X] I have updated the app to version **[0.16.1](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by Xori71** [comment]
Also, I forgot to mention this: after the user sets the days left before the next update, the manga would get updated after those days, and only then would the smart update schedule change to the difference of the set date and last chapter upload date.

**Comment by Kladki** [comment]
I have no idea what sort of algorithm Mihon currently uses to predict schedules, but it seems to be easily thrown off during hiatuses. A simple algorithm that doesn't have this issue is to just look at the spacing between the most recent `n` chapters, and just use the median spacing for the estimate.

**Comment by MajorTanya** [comment]
> I have no idea what sort of algorithm Mihon currently uses to predict schedules, but it seems to be easily thrown off during hiatuses. A simple algorithm that doesn't have this issue is to just look at the spacing between the most recent `n` chapters, and just use the median spacing for the estimate.

Mihon checks the past few uploads and gets the average (or mean, or median, I don't recall precisely).

**Comment by Kladki** [comment]
I believe it is using the mean (can't find where in the code though). Just changing this from mean to median would make it soooo much better. 🙏

**Comment by Kladki** [comment]
Yep, it does use the mean: https://github.com/mihonapp/mihon/blob/7dd595f16e2f484bd46f925c03b77675918208fc/domain/src/main/java/tachiyomi/domain/manga/interactor/FetchInterval.kt#L69-L84
